src/shogun/features/DenseFeatures.h

```diff
@@ -541,6 +543,23 @@ template<class ST> class CDenseFeatures: public CDotFeatures
 	void init();
 
 protected:
+	/*
+	 * Helper method which copies the working feature matrix into the pre-allocated
```

Comment:
I would prefer is this method would not get involved with any re-allocation business (sep. of concerns), but rather only REQUIRE that the sizes match.